docs/docs/render-props.md

```diff
@@ -0,0 +1,224 @@
+---
+id: render-props
+title: Render Props
+permalink: docs/render-props.html
+---
+
+The term ["render prop"](https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce) refers to a simple technique for sharing code between React components using a prop whose value is a function.
+
+## Use Render Props for Cross-Cutting Concerns
+
+Components are the primary unit of code reuse in React, but it's not always obvious how to share the state or behavior that one component encapsulates to other components that need that same state.
+
+For example, the following component tracks the mouse position in a web app:
+
+```js
+class MouseTracker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+    this.state = { x: 0, y: 0 };
+  }
+
+  handleMouseMove(event) {
+    this.setState({
```

Comment:
Did you passed an object to `setState` to set the state asynchronously on purpose or could you have passed a function ?

Reply:
I was going to point that out as well, but thought that the complexity of passing a function might take away from the point of the docs.

Reply:
I could've used a function, but I usually don't do that unless the new state depends on some previous state value. Since the purpose of this doc is not to make any points about asynchrony and `setState`, I think the current implementation is fine.